flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/schema/TableSourceTable.scala

```diff
@@ -50,7 +53,8 @@ class TableSourceTable[T](
     statistic: FlinkStatistic,
     val tableSource: TableSource[T],
     val isStreamingMode: Boolean,
-    val catalogTable: CatalogTable)
+    val catalogTable: CatalogTable,
+    val tableIdentifier: Option[ObjectIdentifier])
```

Comment:
I think `names` already played this role

Reply:
Use `names` is quite hack:
```
if (names.size() == 3) {
  // it has ObjectIdentifier
  return ObjectIdentifier.of(names.get(0), names.get(1), names.get(2));
} else {
  // there is no ObjectIdentifier.
}
```
So I think in future, we should use ObjectIdentifier always instead of names.

Reply:
Then i don't think it will be an `Option`

Reply:
Now, only one way should go to here without `ObjectIdentifier`.
It is temporary table with table api(TableSourceQueryOperation). But I can try to fix it.